compose/composer.go

```diff
@@ -2,50 +2,82 @@ package compose
 
 import (
 	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"time"
 )
 
+const lenSpans = 7
+
 func Compose(numErrors, numTransactions, numSpans, numFrames int) []byte {
+	rand.Seed(time.Now().UnixNano())
 	var buf bytes.Buffer
 
 	buf.Write(Metadata)
 	buf.WriteByte('\n')
 
-	stacktrace := make([]byte, len(StacktraceFrame))
-	copy(stacktrace, StacktraceFrame)
-	frames := multiply([]byte(`"stacktrace"`), stacktrace, numFrames)
-
-	span := make([]byte, len(SingleSpan))
-	copy(span, SingleSpan)
-	span = bytes.Replace(span, []byte(`"stacktrace": [],`), frames, -1)
+	for i := 0; i < numTransactions; i++ {
+		ev := randomized("transaction", 8)
+		write(&buf, ndjsonWrapObj("transaction", ev))
 
-	transaction := ndjsonWrapObj("transaction", SingleTransaction)
-	span = ndjsonWrapObj("span", span)
+		for i := 0; i < numSpans; i++ {
+			ev := randomized(fmt.Sprintf("span%d", rand.Intn(lenSpans)+1), 8)
+			span := make([]byte, len(ev))
+			copy(span, ev)
+			span = bytes.Replace(span, []byte(`"stacktrace":[],`), stacktrace(numFrames), -1)
+			write(&buf, ndjsonWrapObj("span", span))
+		}
+	}
 
-	for i := 0; i < numTransactions; i++ {
-		ndJsonRepeat(&buf, transaction, 1)
-		ndJsonRepeat(&buf, span, numSpans)
+	for i := 0; i < numErrors; i++ {
+		ev := randomized("error", 16)
+		errEvent := make([]byte, len(ev))
+		copy(errEvent, ev)
+		errEvent = bytes.Replace(errEvent, []byte(`"stacktrace":[],`), stacktrace(numFrames), -1)
+		write(&buf, ndjsonWrapObj("error", errEvent))
 	}
 
-	errEvent := make([]byte, len(SingleError))
-	copy(errEvent, SingleError)
-	errEvent = bytes.Replace(errEvent, []byte(`"stacktrace": [],`), frames, -1)
+	return bytes.TrimSpace(buf.Bytes())
+}
 
-	errEvent = ndjsonWrapObj("error", errEvent)
-	ndJsonRepeat(&buf, errEvent, numErrors)
+func randomized(key string, idN int) []byte {
+	f, err := ioutil.ReadFile(fmt.Sprintf("./compose/%s.json", key))
+	if err != nil {
+		panic(err)
+	}
+	var ev map[string]interface{}
+	json.Unmarshal([]byte(f), &ev)
+	ev["id"] = randHexString(idN)
+	ev["transaction_id"] = randHexString(8)
+	ev["parent_id"] = randHexString(8)
+	ev["trace_id"] = randHexString(16)
+	b, err := json.Marshal(ev)
+	if err != nil {
```

Comment:
i'd do the error check in the `Unmarshal` function instead of `Marshal`: 
if there is something wrong with the json file, `Unmarshal` will fail. OTOH, if `Unmarshal` did not fail, `Marshal` wont fail neither